HLTrigger/special/src/HLTPixelActivityHFSumEnergyFilter.cc

```diff
@@ -0,0 +1,129 @@
+  virtual bool filter(edm::Event&, const edm::EventSetup&) override; 
+  virtual void endStream() override;
+  
+  edm::InputTag inputTag_;          // input tag identifying product containing pixel clusters
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster> > inputToken_;
+  edm::EDGetTokenT<HFRecHitCollection> HFHitsToken_;
+  edm::InputTag HFHits_;
+  double eCut_HF_;
+  double eMin_HF_;
+  double offset_;
+  double slope_;
+};
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+
+//
+// constructors and destructor
+//
+
+HLTPixelActivityHFSumEnergyFilter::HLTPixelActivityHFSumEnergyFilter(const edm::ParameterSet& config)
+{
+  inputTag_      = config.getParameter<edm::InputTag>("inputTag");
+  HFHits_        = config.getParameter<edm::InputTag>("HFHitCollection");
+  eCut_HF_       = config.getParameter<double>("eCut_HF");
+  eMin_HF_       = config.getParameter<double>("eMin_HF");
+  offset_        = config.getParameter<double>("offset");
+  slope_         = config.getParameter<double>("slope");
+
+  inputToken_ = consumes<edmNew::DetSetVector<SiPixelCluster> >(inputTag_);
+  HFHitsToken_ = consumes<HFRecHitCollection>(HFHits_); 
+}
+
+HLTPixelActivityHFSumEnergyFilter::~HLTPixelActivityHFSumEnergyFilter()
+{
+}
+
+void
+HLTPixelActivityHFSumEnergyFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("inputTag",edm::InputTag("hltSiPixelClusters"));
+  desc.add<edm::InputTag>("HFHitCollection",edm::InputTag("hltHfreco"));
+  desc.add<double>("eCut_HF",0);
+  desc.add<double>("eMin_HF",0);
+  desc.add<double>("offset",0);
+  desc.add<double>("slope",0);
+  descriptions.add("hltPixelActivityHFSumEnergyFilter",desc);
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+bool HLTPixelActivityHFSumEnergyFilter::filter(edm::Event& event, const edm::EventSetup& iSetup) {
+
+  using namespace edm;
+  // get hold of products from Event
+  edm::Handle<edmNew::DetSetVector<SiPixelCluster> > clusterColl;
+  event.getByToken(inputToken_, clusterColl);
+
+  unsigned int clusterSize = clusterColl->dataSize();
+  LogDebug("") << "Number of clusters accepted: " << clusterSize;
+
+  edm::Handle<HFRecHitCollection> HFRecHitsH;
+  event.getByToken(HFHitsToken_,HFRecHitsH);
+
+  double sumE = 0.;
+  for (HFRecHitCollection::const_iterator it=HFRecHitsH->begin(); it!=HFRecHitsH->end(); it++) {
+      if (it->energy()>eCut_HF_) {    
+          sumE += it->energy();
+      }
+  }
+
+  bool accept = false;
+  double thres = slope_ * sumE - offset_;
+  if(sumE > eMin_HF_ && clusterSize < thres )
+  {
```

Comment:
instead of clusterSize < thres better std::abs(clusterSize - thres) > someValue 
to check deviation from (alomst) equality?